crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -579,6 +552,135 @@ impl ArtifactFetcher {
             sourcemap_url: sourcemap_url.map(Arc::new),
         })
     }
+
+    /// Queries the Sentry API for a single file (by its [`DebugId`] and file stem).
+    async fn query_sentry_for_file(&mut self, key: &FileKey) {
+        let mut debug_ids = BTreeSet::new();
+        if let Some(debug_id) = key.debug_id() {
+            debug_ids.insert(debug_id);
+        }
+
+        let mut file_stems = BTreeSet::new();
+        if let Some(url) = key.abs_path() {
+            let stem = extract_file_stem(url);
+            file_stems.insert(stem);
+        }
+
+        self.query_sentry_for_files(debug_ids, file_stems).await
+    }
+
+    /// Queries the Sentry API for a bunch of [`DebugId`]s and file stems.
+    ///
+    /// This will also download all referenced [`ArtifactBundle`]s directly and persist them for
+    /// later access.
+    /// Individual files are not eagerly downloaded, but their metadata will be available.
+    async fn query_sentry_for_files(
+        &mut self,
+        debug_ids: BTreeSet<DebugId>,
+        file_stems: BTreeSet<&str>,
+    ) {
+        // TODO(sourcemap): this is where we want to hook in the not-yet-existing Sentry API
```

Comment:
Is the intention that the API on the Sentry side returns a minimal set of `FileResult`s containing everything we ask for?